Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb

```diff
@@ -261,7 +275,10 @@ def list_all_blobs
       # Need to limit the returned number of the returned entries to avoid out of memory exception.
       entries = @azure_blob.list_blobs(@container, { :timeout => 60, :marker => continuation_token, :max_results => @blob_list_page_size })
       entries.each do |entry|
-        blobs << entry
+        # FNM_PATHNAME is required so that "**/test" can match "test" at the root folder
+        # FNM_EXTGLOB allows you to use "test{a,b,c}" to match either "testa", "testb" or "testc" (closer to shell behavior)
+        matched = @path.any? {|path| File.fnmatch?(path, entry.name, File::FNM_PATHNAME | File::FNM_EXTGLOB)}
```

Comment:
It may be good to default path to empty string instead of just **/*, and move line 147 for adding the registry path to here.

That way, if the filter is empty, then you can just do blobs << entry, without running File.fnmatch for every single item. The reason why this is better is no performance impact on everyone else today who doesn't have this filter.